src/shapes/Object/Object.ts

```diff
@@ -490,7 +472,10 @@ export class FabricObject<EventSpec extends ObjectEvents = ObjectEvents>
    * @param {Array} [propertiesToInclude] Any properties that you might want to additionally include in the output
    * @return {Object} Object representation of an instance
    */
-  toObject(propertiesToInclude?: string[]): Record<string, any> {
+  toObject<
+    T extends Omit<Props & TClassProperties<this>, keyof SProps>,
+    K extends keyof T = never
+  >(propertiesToInclude?: K[]): { [R in K]: T[K] } & SProps {
```

Comment:
This thing is unreadable and who invented typescript is responsible for my eyes bleeding

Reply:
It was not easy to write and definately not to test